react/src/components/GeneSearch.tsx

```diff
@@ -1,79 +1,66 @@
 import React, { useEffect, useState } from 'react';
 import { useAsyncDebounce } from 'react-table';
-import { useFetchAutocompleteQuery, useFetchEnsemblIdQuery } from '../apollo/hooks';
+import { useFetchAutocompleteQuery } from '../apollo/hooks';
 import { DropdownItem } from '../types';
 import ComboBox from './ComboBox/ComboBox';
 
-export interface GeneOption {
+export type GeneOption = DropdownItem<SelectionValue>;
+
+interface SelectionValue {
     ensemblId: string;
     name: string;
 }
 
 interface GeneSearchProps {
-    value: GeneOption;
+    selectedGene: string;
     onSelect: (gene: GeneOption) => void;
-    onSearch: (name: string) => void;
 }
 
-const GeneSearch: React.FC<GeneSearchProps> = ({ onSearch, onSelect, value: { name } }) => {
-    const [options, setOptions] = useState<DropdownItem[]>([]);
+const GeneSearch: React.FC<GeneSearchProps> = ({ onSelect, selectedGene }) => {
+    const [options, setOptions] = useState<GeneOption[]>([]);
     const [searchTerm, setSearchTerm] = useState('');
-    const [selectedTerm, setSelectedTerm] = useState('');
 
     const [fetchAutocompleteResults, { data: autocompleteResults, loading: autocompleteLoading }] =
         useFetchAutocompleteQuery();
 
-    const [fetchEnsemblIdResults, { data: ensemblIdResults, loading: ensemblIdLoading }] =
-        useFetchEnsemblIdQuery();
-
     const debouncedAutocompleteFetch = useAsyncDebounce(fetchAutocompleteResults, 500);
 
     useEffect(() => {
         if (!searchTerm) {
             setOptions([]);
         }
         if (searchTerm.length > 2) {
-            debouncedAutocompleteFetch({ variables: { term: searchTerm.toLowerCase() } });
+            debouncedAutocompleteFetch({ variables: { q: searchTerm.toLowerCase() } });
         }
     }, [debouncedAutocompleteFetch, searchTerm]);
 
     useEffect(() => {
         if (autocompleteResults) {
             setOptions(
-                (autocompleteResults.autocompleteResults.suggestions || []).map(
-                    (s: { suggestion: string }, i: number) => ({
-                        value: s.suggestion.toUpperCase(),
-                        id: i,
-                        label: s.suggestion.toUpperCase(),
-                    })
-                )
+                (autocompleteResults.autocompleteResults.hits || []).map((hit, i) => ({
+                    value: {
+                        name: hit.symbol.toUpperCase(),
+                        ensemblId: hit.ensembl?.gene,
```

Comment:
Because the ensemblId can be `undefined` here, the program throws an error at this line [here](https://github.com/ccmbioinfo/ssmp/blob/4b92a0f8961a58ec7240cd4241dcd508f53bf299/react/src/pages/VariantQueryPage.tsx#L33). An example is when the gene APOC4-APOC2 is selected, its ensemblId is undefined.